sirepo_flyer.py

```diff
@@ -0,0 +1,241 @@
+            datum_id = new_uid()
+            date = datetime.datetime.now()
+            srw_file = str(Path('/tmp/data') / Path(date.strftime('%Y/%m/%d')) / Path('{}.dat'.format(datum_id)))
+            self._srw_files.append(srw_file)
+            _resource_uid = new_uid()
+            resource = {'spec': 'SIREPO_FLYER',
+                        'root': '/tmp/data',  # from 00-startup.py (added by mrakitin for future generations :D)
+                        'resource_path': srw_file,
+                        'resource_kwargs': {},
+                        'path_semantics': {'posix': 'posix', 'nt': 'windows'}[os.name],
+                        'uid': _resource_uid}
+            self._resource_uids.append(_resource_uid)
+            self._asset_docs_cache.append(('resource', resource))
+
+        for param in self.params_to_change:
+            # for i in range(self.copy_count):
+            # name doesn't need to be unique, server will rename it
+            c1 = sb.copy_sim('{} Bluesky'.format(sb.data['models']['simulation']['name']), )
+            print('copy {}, {}'.format(c1.sim_id, c1.data['models']['simulation']['name']))
+            # want to be able to change any parameters to any value
+
+            for key, parameters_to_update in param.items():
+                optic_id = sb.find_optic_id_by_name(key)
+                c1.data['models']['beamline'][optic_id].update(parameters_to_update)
+            watch = sb.find_element(c1.data['models']['beamline'], 'title', self.watch_name)
+            c1.data['report'] = 'watchpointReport{}'.format(watch['id'])
+            self._copies.append(c1)
+            print('COPIES', len(self._copies))
+
+        if self.run_parallel:
+            procs = []
+            for i in range(self.copy_count):
+                p = Process(target=self._run, args=(self._copies[i],))
+                p.start()
+                procs.append(p)
+            # wait for procs to finish
+            for p in procs:
+                p.join()
+        else:
+            # run serial
+            for i in range(self.copy_count):
+                print('running sim: {}', self._copies[i].sim_id)
+                status = self._copies[i].run_simulation()
+                print('Status:', status['state'])
+        return NullStatus()
+
+    def complete(self, *args, **kwargs):
+        for i in range(len(self._copies)):
+            datum_id = self._resource_uids[i]
+            datum = {'resource': self._resource_uids[i],
+                     'datum_kwargs': {},
+                     'datum_id': datum_id}
+            self._asset_docs_cache.append(('datum', datum))
+            self._datum_ids.append(datum_id)
+        return NullStatus()
+
+    def describe_collect(self):
+        elem_name = []
+        curr_param = []
+        for i in range(len(self.params_to_change)):
+            inputs = self.params_to_change[i]
```

Comment:
I think we can just loop over `self.params_to_change` and get `inputs` in the for loop.